src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,54 @@
+const ADD_BOOK = 'bookstore/books/ADD_BOOK';
+const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
+const TOGGLE_BOOK = 'bookstore/books/TOGGLE_BOOK';
+const initialState = [];
+
+const booksReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return [...state, action.payload];
+    case TOGGLE_BOOK:
+      return state.map((book) => {
+        if (book.id === action.payload) {
+          return {
+            ...book,
+            progress: book.progress === 'read' ? 'currently reading' : 'read',
+          };
+        }
```

Comment:
- [ ] [OPTIONAL] The code you provided is an implementation of Redux logic using Redux Toolkit conventions. However, there are a few improvements and adjustments that can be made to align it more closely with Redux Toolkit's features and best practices. I'll go through the code step by step and provide suggestions for improvement:

```js
import { createSlice } from '@reduxjs/toolkit';

const initialState = [];

const booksSlice = createSlice({
  name: 'books',
  initialState,
  reducers: {
    addBook: (state, action) => {
      state.push(action.payload);
    },
    removeBook: (state, action) => {
      return state.filter(book => book.id !== action.payload);
    },
    toggleBook: (state, action) => {
      const book = state.find(book => book.id === action.payload);
      if (book) {
        book.progress = book.progress === 'read' ? 'currently reading' : 'read';
      }
    },
  },
});

export const { addBook, removeBook, toggleBook } = booksSlice.actions;

export default booksSlice.reducer;
```

- [ ] [OPTIONAL] Please consider doing same for the `categoriesSlice.js` as well

Reply:
@codepantha Thank you for your feedback, I appreciate it.